### PR TITLE
Definition of Metrics API & NOOP implementation

### PR DESCRIPTION
**Pull request by hendrikebbers**
First metrics API

---
This change is [Reviewable](https://reviewable.io/reviews/canoo/dolphin-platform/865)

### CODE REVIEW

File: platform-extras/dolphin-platform-metrics-server/src/main/java/com/canoo/dp/impl/platform/server/metrics/noop/NoopCounter.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.canoo.dp.impl.platform.server.metrics.noop;
+
+import com.canoo.dp.impl.platform.server.metrics.AbstractMeter;
+import com.canoo.dp.impl.platform.server.metrics.MeterIdentifier;
+import com.canoo.platform.metrics.types.Counter;
+
+public class NoopCounter extends AbstractMeter implements Counter {
+
+    public NoopCounter(final MeterIdentifier identifier, AutoCloseable closeable) {
```
Comment (giftkugel) [inline platform-extras/dolphin-platform-metrics-server/src/main/java/com/canoo/dp/impl/platform/server/metrics/noop/NoopCounter.java]: Add `final` keyword.

File: platform-extras/dolphin-platform-metrics-server/src/main/java/com/canoo/dp/impl/platform/server/metrics/noop/NoopGauge.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package com.canoo.dp.impl.platform.server.metrics.noop;
+
+import com.canoo.dp.impl.platform.server.metrics.AbstractMeter;
+import com.canoo.dp.impl.platform.server.metrics.MeterIdentifier;
+import com.canoo.platform.metrics.types.Gauge;
+
+public class NoopGauge extends AbstractMeter implements Gauge {
+
+    public NoopGauge(final MeterIdentifier identifier, AutoCloseable closeable) {
```
Comment (giftkugel) [inline platform-extras/dolphin-platform-metrics-server/src/main/java/com/canoo/dp/impl/platform/server/metrics/noop/NoopGauge.java]: Add `final` keyword.

File: platform-extras/dolphin-platform-metrics-server/src/main/java/com/canoo/dp/impl/platform/server/metrics/noop/NoopTimer.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import com.canoo.dp.impl.platform.server.metrics.AbstractMeter;
+import com.canoo.dp.impl.platform.server.metrics.MeterIdentifier;
+import com.canoo.platform.metrics.types.Timer;
+
+import java.util.concurrent.TimeUnit;
+
+public class NoopTimer extends AbstractMeter implements Timer {
+
+    public NoopTimer(final MeterIdentifier identifier, AutoCloseable closeable) {
```
Comment (giftkugel) [inline platform-extras/dolphin-platform-metrics-server/src/main/java/com/canoo/dp/impl/platform/server/metrics/noop/NoopTimer.java]: Add `final` keyword.

File: platform-extras/dolphin-platform-metrics-server/src/main/java/com/canoo/dp/impl/platform/server/metrics/noop/NoopCounter.java (hunk 1 above)
Comment (hendrikebbers) [reply platform-extras/dolphin-platform-metrics-server/src/main/java/com/canoo/dp/impl/platform/server/metrics/noop/NoopCounter.java]: done

File: platform-extras/dolphin-platform-metrics-server/src/main/java/com/canoo/dp/impl/platform/server/metrics/noop/NoopGauge.java (hunk 2 above)
Comment (hendrikebbers) [reply platform-extras/dolphin-platform-metrics-server/src/main/java/com/canoo/dp/impl/platform/server/metrics/noop/NoopGauge.java]: done

File: platform-extras/dolphin-platform-metrics-server/src/main/java/com/canoo/dp/impl/platform/server/metrics/noop/NoopTimer.java (hunk 3 above)
Comment (hendrikebbers) [reply platform-extras/dolphin-platform-metrics-server/src/main/java/com/canoo/dp/impl/platform/server/metrics/noop/NoopTimer.java]: Done

**Pull request merged**